src/LetsEncrypt.Azure/AzureKeyVaultCertificateRepositoryOptions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Nate McMaster.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Azure.Core;
+using Azure.Identity;
+
+#if FEATURE_VALIDATE_DATA_ANNOTATIONS
+using System.ComponentModel.DataAnnotations;
+#endif
+
+namespace McMaster.AspNetCore.LetsEncrypt
+{
+    /// <summary>
+    /// Options to connect to an Azure KeyVault
+    /// </summary>
+    public class AzureKeyVaultCertificateRepositoryOptions
+    {
+        /// <summary>
+        /// Gets or sets the Url for the KeyVault instance.
+        /// </summary>
+#if FEATURE_VALIDATE_DATA_ANNOTATIONS
+        [Url]
+        [Required]
+#endif
+        public string Url { get; set; } = null!;
```

Comment:
```suggestion
        public string? Url { get; set; }
```
Does this work? Or what about something longer, but with an API that explains the URI's purpose?
```suggestion
        public Uri? AzureKeyVaultEndpoint { get; set; }
```

Reply:
Updated